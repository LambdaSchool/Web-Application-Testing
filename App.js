import React, { useState } from 'react';
import Display from './components/Display.js'
import Dashboard from './components/Dashboard.js'
import './App.css';

export const addStrike = currentStrikes => {
  return currentStrikes + 1;
}

export const addBall = currentBalls => {
  return currentBalls + 1;
}

function App() {
  const [ strikes, setStrikes ] = useState(0);
  const [ balls, setBalls ] = useState(0);

  const handleStrike = () => {
    if(strikes < 2)
      setStrikes(addStrike(strikes));
    else
      setStrikes(0);
  };

  const handleFoul= () => {
    if(strikes < 2)
      setStrikes(strikes + 1);
  };

  const handleBall = () => {
    if(balls < 3)
      setBalls(addBall(balls));
    else
      setBalls(0);
  };

  const handleHit = () => {
      setBalls(0);
      setStrikes(0);
  };


  return (
    <div className="App">
      <Display strikes={strikes} balls={balls}/>
      <Dashboard handleStrike={handleStrike} handleFoul={handleFoul} handleBall={handleBall} handleHit={handleHit}/>
    </div>
  );
}

export default App;
