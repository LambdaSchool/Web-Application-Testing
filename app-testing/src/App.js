import React, { useState } from 'react';
import logo from './logo.svg';

import Display from './components/Display.js';
import Dashboard from './components/Dashboard.js';
import './App.css';

function App() {
  
  const [strikes, setStrikes] = useState(0);
  const [balls, setBalls] = useState(0);
  const [outs, setOuts] = useState(0);

  
  const displayReset = () => {
    setStrikes(0);
    setBalls(0);
  }
  
  const addStrikes = () => {
    if(strikes === 2 && outs === 2){
      clear()
    } else if(strikes === 2){
      displayReset()
      setOuts(outs + 1)
    } else {
      return setStrikes(strikes + 1)
    }
  }
  

  const addBall = () => {
    if(balls === 3){
      setBalls(0)
      displayReset()
    } else {
      return setBalls(balls + 1)
    }
  }
  
  const addFoul = () => {
    if(strikes === 2){
      return strikes;
    } else {
      return setStrikes(strikes + 1)
    }
  }

  const addOut = () => {
    if(outs === 2){
      setOuts(0);
      displayReset();
    } else {
      setOuts(outs + 1);
      displayReset();
    }
  }

  const clear = () => {
    displayReset();
    setOuts(0);
  }

  
  
  return (
    <div className="App">
      <Display strikes={strikes} 
               balls={balls}
               outs={outs}
      />
      <Dashboard addStrike={addStrikes}
                 addBall={addBall}
                 addFoul={addFoul}
                 addOut={addOut}
                 reset={displayReset}
                 clear={clear}
      />
    </div>
  );
}

export default App;
