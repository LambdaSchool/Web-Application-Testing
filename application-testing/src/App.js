import React, { useState } from 'react';
import './App.css';

function App() {
	const [ balls, setBalls ] = useState(0);
	const [ strikes, setStrikes ] = useState(0);

	return (
		<div className="App">
			<Dashboard balls={balls} setBall={setBalls} strikes={strikes} setStrikes={setStrikes} />
			<Display balls={balls} setBall={setBalls} strikes={strikes} setStrikes={setStrikes} />
		</div>
	);
}

export default App;
