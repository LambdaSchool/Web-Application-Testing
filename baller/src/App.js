import React, { Component } from 'react';
import './App.css';
import Display from './components/Display';

class App extends Component {
  render() {
    console.log(this.state);
    return (
      <div className="App">
        <header className="App-header">
          <h1>Big Baller</h1>
          <Display />
        </header>
      </div>
    );
  }
}

export default App;