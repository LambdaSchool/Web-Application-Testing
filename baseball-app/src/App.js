import React from 'react';
import './App.css';


import Display from "./components/Display";
import Dashboard from "./components/Dashboard";


function App() {

  const [balls, setBalls ] = useState(0);
  const [strikes, setStrikes] = useState(0)


  }
  return (
    <div className="App">
      <h1>I Hate Baseball</h1>
      <Display />
      <Dashboard />
    </div>
  );
}

export default App;