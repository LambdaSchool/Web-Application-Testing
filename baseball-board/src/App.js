import React, { useState } from "react";
import Dashboard from './components/Dashboard/Dashboard';
import Display from './components/Display/Display';
import logo from './logo.svg';
import './App.css';

export const addStrike = (count) => {
  return count + 1;
}

export const addFoul = (count) => {
  if (foul < 2 && strike < 2) {
    return count + 1;
  } 
}

export const addBall = (count) => {
  if (foul < 2 && strike < 2) {
    return count + 1;
  }
}

export const addHit = (count) => {
  if (foul < 2 && strike < 2) {
      return count + 1;
  }
}

function App() {

  const [strike, setStrike] = useState(0);
  const [ball, setBall] = useState(0);
  const [foul, setFoul] = useState(0);
  const [hit, setHit] = useState(0);

  return (
    <div className="App">
      {/* <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header> */}
      <Display strike={strike} ball={ball}/>
      <Dashboard strike={strike} setStrike={setStrike} ball={ball} 
      setBall={setBall} foul={foul} setFoul={setFoul} hit={hit} 
      setHit={setHit}/>
    </div>
  );
}

export default App;
