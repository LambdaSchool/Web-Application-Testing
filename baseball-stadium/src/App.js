import React from 'react';
import './App.css';
import Display from "./Components/Display";
import Dashboard from "./Components/Dashboard";

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <h1>"Baseball Stadium" Project</h1>
        <Display />
        <Dashboard />
      </header>
    </div>
  );
}

export default App;
