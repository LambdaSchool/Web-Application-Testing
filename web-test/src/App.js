import React, { useState } from 'react';
import './App.css';
import Dashboard from './components/dashboard';
import Display from './components/display';
function App() {
  const [strikes, setStrikes] = useState(0);
  const [balls, setBalls] = useState(0);
  return (
    <div className="App">
      <h1>At bat baseball tracker</h1>
      <Dashboard strikes={strikes} setStrikes={setStrikes} balls={balls} setBalls={setBalls} />
      <Display strikes={strikes} balls={balls} setStrikes={setStrikes}setBalls={setBalls}  />
    </div>
  );
}

export default App;
