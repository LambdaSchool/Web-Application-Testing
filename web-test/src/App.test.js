import React from 'react';
import "@testing-library/jest-dom/extend-expect";
import { render, fireEvent, cleanup } from '@testing-library/react';
import App from './App';
// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
afterEach(cleanup);
it('matches snapshot', () =>{
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

it('renders 4 buttons w/o crashing', ()=>{
  const { getByTestId } = render(<App/>);
  const buttonDiv = getByTestId('btnParent');
  expect(buttonDiv.children.length).toBe(4);
})