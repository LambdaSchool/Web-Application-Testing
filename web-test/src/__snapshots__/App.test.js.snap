// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      At bat baseball tracker
    </h1>
    <div
      data-testid="btnParent"
    >
      <button>
        Strike
      </button>
      <button>
        Ball
      </button>
      <button>
        Foul
      </button>
      <button>
        Hit
      </button>
    </div>
    <div>
      <h2>
        Strikes: 0
      </h2>
      <h2>
        Balls: 0
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
